src/checkin/services/flow/step/GetExtraPassengerInformationStep.ts

```diff
@@ -0,0 +1,42 @@
+import { Builder } from 'builder-pattern'
+import { Context } from '../../../model/Context'
+import { Session } from '../../../model/session'
+import { UserInformation } from '../../../model/session/UserInformation'
+import StepTemplate from '../StepTemplate'
+import { SessionData } from '../../../model/session/SessionData'
+
+export default class GetExtraPassengerInformationStep extends StepTemplate {
+
+  when(context: Context): boolean {
+    const numberOfPassengers = context.getSession().data.passengers
+    return numberOfPassengers > 1
+  }
+
+  onExecute(context: Context): Promise<boolean> {
+    const session = context.getSession()
+    const requestData = context.getRequest()
+    const amountOfExtraPassengers = session.data.passengers - 1
+    const extraPassengerInformation = requestData.fields?.extra_passenger_information as Array<{ name: string, passport: string }>
+
+    if (!extraPassengerInformation || amountOfExtraPassengers > extraPassengerInformation.length) {
+      context = context.withResponseBuilder((responseBuilder) => responseBuilder.status('user_information_required').requiredFiles({ extra_passenger_information: null }))
+      return Promise.resolve(false)
+    }
+
+    context = this.setExtraPassengerInformationInSession(context, session, extraPassengerInformation)
+
+    return Promise.resolve(true)
+  }
+
+  private setExtraPassengerInformationInSession(context: Context, session: Session, extraPassengerInformation: Array<{ name: string, passport: string }>): Context {
+    context = context.withSessionBuilder((sessionBuilder) => sessionBuilder
+      .userInformation(Builder<UserInformation>(session.userInformation).build())
+      .data(Builder<SessionData>(session.data).build())
+      .extra_passenger_information(extraPassengerInformation)
+      .sessionId(session.sessionId)
+      .userId(session.userId))
```

Comment:
The withBuilder is made in the way that you can only pass the required field you want to add. it is not necesary to set again all the information in the session.
You can just do:
```
context = context.withSessionBuilder((sessionBuilder) => sessionBuilder
      .extra_passenger_information(extraPassengerInformation))
```